### PR TITLE
Crentential support

### PR DESCRIPTION
**Pull request by elgorro**
-This should fix an error 401 when security is credentials.
-Tested locally with Win10 IIS and only win auth enabled also the "no security" works with google. Can't tetst basic
-I would appreciate a merge, but feel free to correct things and optimazation (Message me if you need help)
-FYI: it's my first pr on github... dunno how this works

### CODE REVIEW

File: website-warmup/task.json [hunk 1]
```diff
@@ -70,18 +70,34 @@
...
+                "cred": "cred",
+                "none": "none"
+            },
+            "helpMarkDown": "Optionally choose an authentification type."
+        },
+        {
+            "name": "User",
             "type": "string",
-            "label": "Basic Auth User",
+            "label": "Username",
```
Comment (elgorro) [inline website-warmup/task.json]: Ops... This should be just **User**... Sry it's late here :)

**Comment by luuksommers** [comment]
Sorry that your change has been created on an older version, I love the idea and I'm happy to integrate this. Can you please rebase your change on the 0.3.8 version (or create a new PR if thats easier). Thank you for your PR!

**Comment by elgorro** [comment]
Ok.. here we go! I set the preview flag, so we can test it. Let me know your thougths. :)

**Comment by elgorro** [comment]
@luuksommers Well I know I implemented a lot of necessary code and but also some gimmicks! Can we go forward somehow?

- I can split these in different PRs of course
- I can change or revert some things

**Comment by luuksommers** [comment]
Sorry for the late reply, I'll check it tonight

File: website-warmup/task.json [hunk 2]
```diff
@@ -13,8 +13,9 @@
     "version": {
         "Major": 1,
         "Minor": 2,
```
Comment (luuksommers) [inline website-warmup/task.json:15]: I would suggest updating Minor to 3 and Patch to 0. When adding a non breaking new feature semver suggests updating the Minor version

File: website-warmup/task.json [hunk 3]
```diff
@@ -70,18 +71,42 @@
...
         {
-            "name": "BasicAuthUser",
+            "name": "Authentication",
+            "type": "pickList",
+            "label": "Select authentication type",
+            "required": false,
+            "defaultValue": "none",
+            "groupName": "auth",
+            "options": {
+                "basic": "basic",
```
Comment (luuksommers) [inline website-warmup/task.json:81]: Can you update this to Basic (capital B)
```suggestion
                "basic": "Basic",
```

File: website-warmup/task.json [hunk 4]
```diff
@@ -70,18 +71,42 @@
...
+                "cred": "win cred",
```
Comment (luuksommers) [inline website-warmup/task.json:82]: Same here
```suggestion
                "cred": "Windows Credentials",
```

File: website-warmup/task.json [hunk 5]
```diff
@@ -70,18 +71,42 @@
...
+                "none": "none"
```
Comment (luuksommers) [inline website-warmup/task.json:83]:
```suggestion
                "none": "None"
```

File: website-warmup/website-warmup.ps1 [hunk 6]
```diff
@@ -51,16 +60,31 @@ if ($ignoreSslError)
...
+            $encodedCreds = [System.Convert]::ToBase64String([System.Text.Encoding]::ASCII.GetBytes($pair))
+            
+            $basicAuthValue = "Basic $encodedCreds"
+            
+            $Headers = @{
+                Authorization = $basicAuthValue
+            }
+        }
+      }
+    "cred" {
```
Comment (luuksommers) [inline website-warmup/website-warmup.ps1]:
```suggestion
    "windowscredentials" {
```

File: website-warmup/website-warmup.ps1 [hunk 7]
```diff
@@ -81,26 +105,36 @@ if(-not $suffixes) {
...
+                throw $_
+            }
         }
     }
 
     if($siteIsAlive){
         Write-Host "Site is running in $($time.TotalSeconds) seconds"
     } else {
-        Write-Host "Site returned error after $retryCount tries and in $($time.TotalSeconds) seconds"
+        Write-Host "Site returned error after $retryCount tries and in $($time.TotalSeconds) seconds!"
```
Comment (luuksommers) [inline website-warmup/website-warmup.ps1:135]: YES! thanks for adding the critical `!` 🤔

**Comment by luuksommers** [comment]
Thank you so much for your change! I'll merge your code and update it myself, you you'll have the preview version available tomorrow! 0.4.0 is uploading now! Sorry for the lonnnggggg delay, here's an octopus: 🐙

**Comment by luuksommers** [comment]
Published 0.4.1, the group was missing.

**Pull request merged**